### ISSUE TITLE
BGP advertisement configuration is confusing, makes it easy to configure 0 advertisements

### ISSUE THREAD

**Issue by xnaveira** [google/metallb#45]
When updating bgp speakers config if no localpref is explicitly declared in the pool config 

```      
advertisements:
      - localpref: 100
```
the configuration is applied but when a new loadbalanced service is created the advertisement will be empty:
https://github.com/google/metallb/blob/6d7a06f095d0a429e551d10fc610919bf8da2398/bgp-speaker/main.go#L111

which makes bgpspeaker simply ignore that advertisement with the message

```
<servicename>: announcable, making 0 advertisements
```

So ip is assigned from the pool but it never gets announced.

There should be a default for localpref or else such a config should trigger an error.

**Comment by danderson** [comment]
I think the problem here is not the lack of localpref, but a confusing aspect of the configuration.

The correct way to request an advertisement with everything set to the defaults is:

```yaml
  advertisements:
  -
```

That creates a list of 1 advertisement, with no settings. If you don't have the `-`, the configuration means "you should make 0 BGP advertisements for this IP range.

This is an unfortunate consequence of how I structured the config file originally. I think we'll change this in 0.3, when @miekg adds ARP announcing support, we'll need a way to explicitly say "I want you to make BGP advertisements", separately from "here's some BGP settings".

**Comment by xnaveira** [comment]
Oh I see, i didn't look at it close enough. I guess if you are going to change it anyway it is fine. From an end user perspective it would be weird that the configuration is accepted but no announcements are being made.

**Comment by danderson** [comment]
Yeah, that was a thing I realized while I was writing the prototype, and then decided that we were going to refactor/change the configuration anyway for ARP and RIP support, so I would just fix it then.

In future, I think the address-pool section will look something like:

```yaml
  name: my-pool
  cidr:
  - 10.0.0.0/8
  advertise-bgp: true # If you say only this, we do something reasonable: advertise individual /32s, localpref=0, no communities.
  bgp-config: # If you want something different than the default, you must specify >=1 element in this section.
  - localpref: 42
    aggregation-length: 24
```

And then when we have arp and rip support, we can add `advertise-arp: true`, `advertise-rip: true`, and `arp-config` and `rip-config` sections if we need them. WDYT? Does that config structure make more sense to you?

**Comment by xnaveira** [comment]
Yeah, I think that would be definitely easier to get right. Just out of curiosity, in which cases do you think an arp proxy would be needed for services?

**Comment by danderson** [comment]
Will be fixed in 0.3.0: if a pool does not specify any BGP advertisements, we use a default that advertises a /32 with no other attributes. That way, there is no way to enable BGP advertising, but accidentally configure zero advertisements.
